lib/util/http/index.ts

```diff
@@ -100,18 +101,22 @@ export class Http<GetOptions = HttpOptions, PostOptions = HttpPostOptions> {
       httpOptions
     );
 
-    if (process.env.NODE_ENV === 'test') {
-      options.retry = 0;
+    if (options) {
+      if (process.env.NODE_ENV === 'test') {
+        options.retry = 0;
+      }
+      options.hooks = {
+        beforeRedirect: [removeAuthorization],
+      };
+    }
+    if (options) {
+      applyDefaultHeaders(options);
     }
-    options.hooks = {
-      beforeRedirect: [removeAuthorization],
-    };
-
-    applyDefaultHeaders(options);
-
     options = applyHostRules(url, options);
-    if (options.enabled === false) {
-      throw new Error(HOST_DISABLED);
+    if (options) {
```

Comment:
Maybe this if here is redundant?

Reply:
after `options = applyHostRules(url, options);` we have to check if `options` contains values or is undefined, so we need to check this object before accessing any of its properties and without this condition, the null-check will fail.